### PR TITLE
Integration test that checks for settings upgrade 

### LINKED ISSUE #670: Add integration test for settings upgrade
**Is your feature request related to a problem? Please describe.**

Coming from https://github.com/opensearch-project/job-scheduler/pull/20 and https://github.com/opensearch-project/opensearch-plugins/pull/15 we want an integration test that sets a legacy setting with a previous version of ODFE/ES and ensures that the upgrade falls back to the previously set setting.

**Describe the solution you'd like**

An integration test, possibly ala https://github.com/opensearch-project/OpenSearch/blob/main/qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/FullClusterRestartSettingsUpgradeIT.java.

**Describe alternatives you've considered**

A settings integ test ala https://github.com/opensearch-project/OpenSearch/blob/main/server/src/internalClusterTest/java/org/opensearch/common/settings/UpgradeSettingsIT.java, but that doesn't really test upgrade.

### PR DESCRIPTION
**Pull request by meghasaik**
Signed-off-by: Megha Sai Kavikondala <kavmegha@amazon.com>

Description
[Describe what this change achieves]
An Integration test which checks for the settings upgrade by setting the legacy settings for the previous version of ODFE/ES in the old cluster and after upgrading it to the new version, checks if the settings fallback to the previously set settings.

Issues Resolved
#670
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by meghasaik** [comment]
start gradle check

File: qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/TestSettingsIT.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            if (isRunningAgainstOldCluster()) {
+                final Request putSettingsRequest = new Request("PUT", "/_cluster/settings");
+                try (XContentBuilder builder = jsonBuilder()) {
+                    builder.startObject();
+                    {
+                        builder.startObject("persistent");
+                        {
+                            builder.field("search.remote.foo.skip_unavailable", true);
+                            builder.field("search.remote.foo.seeds", Collections.singletonList("localhost:9200"));
+                            builder.field("search.remote.foo.proxy", "localhost:9200");
```
Comment (saratvemulapalli) [inline qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/TestSettingsIT.java:66]: Trying to understand, are these settings deprecated in newer versions of OpenSearch?

Comment (meghasaik) [reply qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/TestSettingsIT.java:66]: In the newer version, the search.remote settings have been  deprecated through 'Setting.Property.Deprecated' value.

Comment (saratvemulapalli) [reply qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/TestSettingsIT.java:66]: Got it.
Looks like we are re-using most of the code from https://github.com/opensearch-project/OpenSearch/blob/main/qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/FullClusterRestartSettingsUpgradeIT.java. 

Isn't this already testing these remote settings and are also deprecated?

**Comment by meghasaik** [comment]
start gradle check

**Review by saratvemulapalli — APPROVED** [review]
Thanks @meghasaik !

**Pull request merged**